scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -546,6 +585,16 @@ def handle(self):
         if hardware.flash_with_test:
             flash_timeout += self.timeout
 
+        harness_name = self.instance.testsuite.harness.capitalize()
+        harness_import = HarnessImporter(harness_name)
+        harness = harness_import.instance
+        harness.configure(self.instance)
+        halt_monitor_evt = threading.Event()
+
+        logger.debug('Flash Before: %d', flash_before)
+        if flash_before:
+            flash_error = self.run_flash(command, flash_timeout, halt_monitor_evt)
+
```

Comment:
1) shouldn't it check `halt_monitor_evt.is_set()` and skip opening Serial and not starting the monitoring thread later ?
2) why `post_flash_script` is not executed here right after the flash operation ?

Reply:
3. with the serial_pty option it will start its `ser_pty_process` before the flash command at this point even with `flash_before`

Reply:
Thanks @golowanow !

Reply:
@golowanow I tried to address the issues you've mentioned. The `post_flash_script` is now executed after `run_flash`. The `monitor_serial` thread  now waits for serial port to be opened, but still can handle `halt_monitor_evt` in case there is a flash error as soon as the serial port is open. 

It is not clear to me the impact on `device-serial-pty`. Is there a problem starting the `ser_pty_process` before flashing the device, even when the serial is attached later?

Thanks!